homeassistant/components/proxmoxve/__init__.py

```diff
@@ -145,41 +103,100 @@ def build_client() -> ProxmoxAPI:
         host_name = host_config["host"]
         coordinators[host_name] = {}
 
-        proxmox_client = hass.data[PROXMOX_CLIENTS][host_name]
+    # import to config flow
+    if DOMAIN in config:
+        for conf in config[DOMAIN]:
+            _LOGGER.warning(
+                # Proxmox VE config flow added in 2022.8 and should be removed in 2022.10
+                "Configuration of the Proxmox in YAML is deprecated and "
+                "will be removed in Home Assistant 2022.10; Your existing configuration "
+                "has been imported into the UI automatically and can be safely removed "
+                "from your configuration.yaml file"
+            )
+            hass.async_create_task(
+                hass.config_entries.flow.async_init(
+                    DOMAIN, context={"source": SOURCE_IMPORT}, data=conf
+                )
+            )
 
-        # Skip invalid hosts
-        if proxmox_client is None:
-            continue
+    return True
 
-        proxmox = proxmox_client.get_api_client()
 
-        for node_config in host_config["nodes"]:
-            node_name = node_config["node"]
-            node_coordinators = coordinators[host_name][node_name] = {}
+async def async_setup_entry(hass: HomeAssistant, config_entry: ConfigEntry) -> bool:
+    """Set up the platform."""
 
-            for vm_id in node_config["vms"]:
-                coordinator = create_coordinator_container_vm(
-                    hass, proxmox, host_name, node_name, vm_id, TYPE_VM
-                )
+    entry_data = config_entry.data
 
-                # Fetch initial data
-                await coordinator.async_refresh()
+    hass.data[PROXMOX_CLIENTS] = {}
 
-                node_coordinators[vm_id] = coordinator
+    host = entry_data[CONF_HOST]
+    port = entry_data[CONF_PORT]
+    user = entry_data[CONF_USERNAME]
+    realm = entry_data[CONF_REALM]
+    password = entry_data[CONF_PASSWORD]
+    verify_ssl = entry_data[CONF_VERIFY_SSL]
 
-            for container_id in node_config["containers"]:
-                coordinator = create_coordinator_container_vm(
-                    hass, proxmox, host_name, node_name, container_id, TYPE_CONTAINER
-                )
+    try:
+        # Construct an API client with the given data for the given host
+        proxmox_client = ProxmoxClient(host, port, user, realm, password, verify_ssl)
+        await hass.async_add_executor_job(proxmox_client.build_client)
```

Comment:
This is duplicated logic from `async_setup`. Don't repeat yourself.

Reply:
Removed duplicates between async_setup and async_setup_entry, leaving async_setup just to import YAML configuration.